docs/services/newrelic.md

```diff
@@ -0,0 +1,58 @@
+# NewRelic
+
+## Parameters
+
+* `apiURL` - the api server url, e.g. https://api.newrelic.com
+* `apiKey` - a [NewRelic ApiKey](https://docs.newrelic.com/docs/apis/rest-api-v2/get-started/introduction-new-relic-rest-api-v2/#api_key)
+
+## Configuration
+
+1. Create a NewRelic [Api Key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#user-api-key)
+2. Store apiKey in `argocd-notifications-secret` Secret and configure NewRelic integration in `argocd-notifications-cm` ConfigMap
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <config-map-name>
+data:
+  service.mattermost: |
+    apiURL: <api-url>
+    apiKey: $newrelic-apiKey
+```
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <secret-name>
+stringData:
+  newrelic-apiKey: apiKey
+```
+
+3. Copy [Application ID](https://docs.newrelic.com/docs/apis/rest-api-v2/get-started/get-app-other-ids-new-relic-one/#apm)
+4. Create subscription for your NewRelic integration
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  annotations:
+    notifications.argoproj.io/subscribe.<trigger-name>.newrelic: <app-id>
+```
+
+## Templates
+
+* `description` - A high-level description of this deployment, visible in the [Summary](https://docs.newrelic.com/docs/apm/applications-menu/monitoring/apm-overview-page) page and on the [Deployments](https://docs.newrelic.com/docs/apm/applications-menu/events/deployments-page) page when you select an individual deployment. Defaults to `message`
+* `changelog` - optional, A summary of what changed in this deployment, visible in the [Deployments](https://docs.newrelic.com/docs/apm/applications-menu/events/deployments-page) page when you select (selected deployment) > Change log
+
+```yaml
+context: |
+  argocdUrl: https://example.com/argocd
+
+template.app-deployed: |
+  message: Application {{.app.metadata.name}} has been healthy.
+  newrelic:
+    description: Application {{.app.metadata.name}} has been healthy
+    changelog: "{{(call .repo.GetCommitMetadata .app.status.sync.revision).Message}}"
```

Comment:
If you not define user you take it from metadata, probably same should be done for changelog ?